spiped/dispatch.c

```diff
@@ -125,6 +127,8 @@ callback_gotconn(void * cookie, int s)
 
 	/* We have gained a connection. */
 	A->nconn += 1;
+	if (A->only_one)
```

Comment:
On second thought, it *is* nice being able to test that cleanup, so I'd rather leave `-1` unchanged.